src/manifest.rs

```diff
@@ -1,17 +1,34 @@
+use chrono::{DateTime, Utc};
+use rocket::http::RawStr;
+use rocket::request::FromFormValue;
 use semver::{Version, VersionReq};
 use std::collections::HashMap;
 
 pub type PackageId = String;
 pub type PackageFullName = String;
 
+#[derive(Serialize, Deserialize, Clone, Eq, PartialEq, Hash, Debug)]
+pub struct FormDateTime(DateTime<Utc>);
```

Comment:
`FormDateTime` should be in `search`. 

It's an implementation detail on how you handle requests which contains dates, and has nothing to do with Manifests directly. It's on a lower level.